3-enrich/stream-enrich/kinesis/src/main/scala/com.snowplowanalytics.snowplow.enrich.stream/sources/KinesisSource.scala

```diff
@@ -140,11 +142,18 @@ class KinesisSource private (
     log.info(s"Processing raw input stream: ${config.in.raw}")
 
     val rawEventProcessorFactory = new RawEventProcessorFactory()
-    val worker = new Worker(
-      rawEventProcessorFactory,
-      kinesisClientLibConfiguration
-    )
-
+    val worker = if (kinesisConfig.disableCloudWatch.isDefined && kinesisConfig.iscloudwatchDisabled == true){
```

Comment:
You can directly use `kinesisConfig.disableCloudWatch` :
```
val worker = kinesisConfig.disableCloudWatch match {
  case Some(true) =>
    new Worker(
      rawEventProcessorFactory,
      kinesisClientLibConfiguration,
      new NullMetricsFactory()
    )
  case _ =>
    new Worker(
      rawEventProcessorFactory,
      kinesisClientLibConfiguration
    )
}
```

Reply:
done